lib/pbkit/pbkit.c

```diff
@@ -2287,163 +2288,101 @@ void pb_end(DWORD *pEnd)
             }
         }
     }
+    return NULL;
 }
 
 
-
-void pb_push1to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1)
+inline DWORD *pb_push_to(DWORD subchannel, DWORD *p, DWORD command, DWORD nparam)
 {
 #ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push1to: new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push1to: missing pb_begin earlier\n");
-    pb_PushIndex+=2;
-    pb_PushNext+=2;
-    if (pb_PushIndex>128) debugPrint("pb_push1to: begin-end block musn't exceed 128 dwords\n");
+    if (p!=pb_PushNext) debugPrint("pb_push_to: new write address invalid or not following previous write addresses\n");
+    if (pb_BeginEndPair==0) debugPrint("pb_push_to: missing pb_begin earlier\n");
+    pb_PushIndex += 1 + nparam;
+    pb_PushNext += 1 + nparam;
+    if (pb_PushIndex>128) debugPrint("pb_push_to: begin-end block musn't exceed 128 dwords\n");
 #endif
 
-    *(p+0)=EncodeMethod(subchannel,command,1);
-    *(p+1)=param1;
+    p=EncodeMethod(subchannel,command,nparam);
```

Comment:
This doesn't look right. `EncodeMethod` doesn't return a pointer

Reply:
Good catch. This change has been reverted. Thanks!